src/App.tsx

```diff
@@ -0,0 +1,68 @@
+import { Route, Routes } from 'react-router';
+import { useMemo } from 'react';
+import BookCardDetails from './components/BookCardDetails/BookCardDetails';
+import Header from './components/Header/Header';
+import Footer from './components/Footer/Footer';
+import { useLocation, useParams } from 'react-router-dom';
+import HomePage from './pages/HomePage/HomePage';
+import { SearchResult } from './pages/SearchResultsPage/SearchResultsPage';
+import {
+  COUNT_PARAM_NAME,
+  PAGE_PARAM_NAME,
+  useStateFromQuery,
+} from './route.utils';
+import ErrorBoundary from './ErrorBoundary';
+import { AppLink } from './components/AppLink';
+
+function useQuery() {
+  const { search } = useLocation();
+  return useMemo(() => new URLSearchParams(search), [search]);
+}
+
+function BindSearchResult() {
+  const { searchTerm } = useParams();
+  const query = useQuery();
+  return (
+    <SearchResult
+      searchTerm={searchTerm ?? ''}
+      page={parseInt(query.get(PAGE_PARAM_NAME) || '0')}
+      limit={parseInt(query.get(COUNT_PARAM_NAME) || '10')}
+    />
+  );
+}
+
+function BindDetailsId() {
+  const queryParams = useStateFromQuery();
+  return (
+    <div>
+      <ErrorBoundary>
+        <BookCardDetails
+          id={queryParams.bookId ?? ''}
```

Comment:
you use empty line, undefined in the code for the `bookId`, probably the `null` is better. But '' also fine. But do not use different values throughout the code